app/services/alert_image_generator.rb

```diff
@@ -129,6 +133,7 @@ def bike_location
 
   # The bike url to be displayed on the promoted alert image
   def bike_url
+    return if bike&.id.blank?
```

Comment:
This happened?

Reply:
I'm not sure how, but there's no FK constraint on this relation, so it can. Probably a cascading delete / destroy is missing:

```rb
StolenRecord.where.not(bike_id: Bike.select(:id)).count
# (312.5ms)  SELECT COUNT(*) FROM "stolen_records" WHERE "stolen_records"."current" = $1 AND ("stolen_records"."bike_id" NOT IN (SELECT "bikes"."id" FROM "bikes" WHERE "bikes"."example" = $2 AND "bikes"."hidden" = $3 AND "bikes"."deleted_at" IS NULL  ORDER BY listing_order desc))  [["current", "t"], ["example", "f"], ["hidden", "f"]]
# => 170


StolenRecord.where.not(bike_id: Bike.select(:id)).map(&:bike).uniq
# [nil]
```

Reply:
Ahhh. Yeah, probably because of `default_scope` stuff. Might be resolved by `bike_id` instead? But doesn't really matter. You've made your point!